src/Pecee/Http/Input/InputHandler.php

```diff
@@ -62,33 +78,46 @@ public function __construct(Request $request)
      */
     public function parseInputs(): void
     {
+        $this->originalParams = [];
+        $this->get = [];
+        $this->originalPost = [];
+        $this->post = [];
+        $this->body_plain = '';
+        $this->originalBody = [];
+        $this->body = [];
+        $this->originalFile = [];
+        $this->file = [];
+
         /* Parse get requests */
         if (\count($_GET) !== 0) {
             $this->originalParams = $_GET;
             $this->get = $this->parseInputItem($this->originalParams);
         }
 
         /* Parse post requests */
-        $this->originalPost = $_POST;
-
-        if (\in_array($this->request->getMethod(), Request::$requestTypesPost, false) === true) {
-
-            $contents = file_get_contents('php://input');
+        if (\count($_POST) !== 0) {
+            $this->originalPost = $_POST;
+            $this->post = $this->parseInputItem($this->originalPost);
+        }
 
-            // Append any PHP-input json
-            if (strpos(trim($contents), '{') === 0) {
-                $post = json_decode($contents, true);
+        /* Get body */
+        $this->body_plain = file_get_contents('php://input');
 
-                if ($post !== false) {
-                    $this->originalPost += $post;
+        /* Parse body */
+        if (in_array($this->request->getMethod(), ['put', 'patch', 'delete', 'post'], false)) {
+            if(strpos($this->request->getContentType(), 'application/json') !== false){
```

Comment:
true and you can extend it better 👍🏻

Reply:
Ah, this is not possible. As I described here:

> It is important to use the contains function, because you can have a content-type like `application/x-www-form-urlencoded; charset=UTF-8`.

Reply:
Nice spottet! Then my suggestion would not work... I've added your `getContentType` method to the `Request` class. Instead of setting it manually I've added so it uses `setContentType` that parses the content-type and strips the charset part so that the result is always uniform. I've also added some generic `CONTENT_TYPE` constants you can use `Request::CONTENT_TYPE_X_FORM_ENCODED` and so on...

Reply:
I already added them 👍🏻